licenses/library.go

```diff
@@ -117,7 +118,33 @@ func Libraries(ctx context.Context, classifier Classifier, includeTests bool, ig
 			klog.Errorf("Package %s does not have module info. Non go modules projects are no longer supported. For feedback, refer to https://github.com/google/go-licenses/issues/128.", p.PkgPath)
 			return false
 		}
-		licensePath, err := Find(pkgDir, p.Module.Dir, classifier)
+
+		stopAt := p.Module.Dir
+		if isVendorModule(newModule(p.Module), pkgDir) {
+			splits := strings.SplitN(pkgDir, "/vendor/", 2)
+			moduleSplits := strings.Split(p.Module.Path, "/")
+			// There seems to be 3 common cases in terms of go module directory structure:
+			// - "github" case follows github.com/<org>/<repo>
+			// - "k8s.io" case follow k8s.io/<module>
+			// - "go.starlark.net" (and probably lots others not exactly sure why/when this happens) case where go.starlark.net is the only folder in the structure
+			// To account for the above, the stopAt is set to the the greatest depth up to 3 subfolders
+			// Alternative: Since license files should only exists in the context of module folders within
```

Comment:
I like this alternative solution. The assumption that vendor folder and module dir folders do not have license files makes sense and is simpler than assuming a depth <= 3.

What do you think?

Reply:
I think so too, we can always adjust it if/when running into a case of a license file being in vendor, which seems less likely than depending on the current assumptions holding true.

Im going to change it in my build and run it thru our pipelines and just make sure nothing new pops up and Ill update this PR.